x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_definitions/entity_descriptions/generic.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { newestValue } from './field_utils';
+import type { EntityDescription } from '../types';
+import { getCommonFieldDescriptions } from './common';
+
+export const GENERIC_DEFINITION_VERSION = '1.0.0';
+export const GENERIC_IDENTITY_FIELD = 'entity.id';
+export const genericEntityEngineDescription: EntityDescription = {
+  entityType: 'generic',
+  version: GENERIC_DEFINITION_VERSION,
+  identityField: GENERIC_IDENTITY_FIELD,
+  settings: {
+    timestampField: '@timestamp',
+  },
+  fields: [
+    newestValue({ source: 'entity.name' }),
+    newestValue({ source: 'entity.source' }),
+    newestValue({ source: 'entity.category' }),
+    newestValue({ source: 'entity.type' }),
+    newestValue({ source: 'entity.address' }),
+
+    newestValue({ source: 'cloud.account.id' }),
+    newestValue({ source: 'cloud.account.name' }),
+    newestValue({ source: 'cloud.availability_zone' }),
+    newestValue({ source: 'cloud.instance.id' }),
+    newestValue({ source: 'cloud.instance.name' }),
+    newestValue({ source: 'cloud.machine.type' }),
+    newestValue({ source: 'cloud.project.id' }),
+    newestValue({ source: 'cloud.project.name' }),
+    newestValue({ source: 'cloud.provider' }),
+    newestValue({ source: 'cloud.region' }),
+    newestValue({ source: 'cloud.service.name' }),
```

Comment:
I though initially that we would be storing only the `entity.*` field in this store as a sort of "Entity" Entity Store? In my understanding, the added fields such as cloud and orchestrator would live in their own entity store definition

Reply:
The current implementation is the means of having as rich as possible entities in the final index. We want to present and give as much context as possible to customers on our flyouts.

This isn't, however, the only solution. We could think on different approaches such as having an entity definition per entity type (maybe ecs field is more accurate) and then joining it on retrieval to build a rich context for the fly out. That raises other concerns regarding the amount of stores, joins and so on.

I think we should discuss and agree on a solution that brings the desired result to the product.

Reply:
Thanks, so as discussed on the previous sync, we are going to move forward with the proposed fields for the generic entity store for 9.1, while we the discussion is in progress about how to leverage the concrete stores for future releases